### ISSUE TITLE
Undesired Removal of Double Square Brackets

### ISSUE THREAD

**Issue by kookma** [flibbles/tw5-uglify#13]
Uglify convert the below macro 

```
\define toc-list(here,exclude)
<$list filter="""[tag[$here$]] $exclude$ -[[$here$]]""">
   <$text text="[["/><<currentTiddler>><$text text="]]"/>
   <$macrocall $name="toc-list" here=<<currentTiddler>> exclude="""$exclude$ -[[$here$]]"""/>
</$list>
\end
```

To this one

```
\define toc-list(here exclude)
<$list filter="""[tag[$here$]]$exclude$ -$here$""">
   <$text text=[[/><<currentTiddler>><$text text=]]/>
   <$macrocall here={{!!title}}exclude="""$exclude$ -[[$here$]]"""$name=toc-list/>

\end
```

The uglified version does not work. Look at the filter of first line:
- original `<$list filter="""[tag[$here$]] $exclude$ -[[$here$]]""">`
- uglified `<$list filter="""[tag[$here$]]$exclude$ -$here$""">`

When $here$ is a title with space the uglified macro fails.

I use this in a TOC macro developed by @ericshulman. Of course, this is an old syntax and text substitution is not recommended anymore. From TW 5.2.x I can use `format:titlelist[]`, so I resolved the issue.

This is just to inform you about the way uglifier works with such macro arguments.

**Comment by flibbles** [comment]
Those placeholder macros are the bane of my existence. If they didn't exist, both Uglify and Relink would only need about 60% of the amount of code they have.

Thanks for reporting this. I don't have time at the moment, but this does need to be fixed eventually.

**Comment by kookma** [comment]
I expect new codes (e.g. TW 5.2 afterwards) do not use such text substitutions.
I avoid using them.

**Comment by flibbles** [comment]
Fixed as of v1.9.1.
